cmd/main.go

```diff
@@ -145,7 +168,43 @@ func parseCliArgs(flags *args) (*cobra.Command, error) {
 	return rootCmd, nil
 }
 
-func run(kubeconfigArgs *genericclioptions.ConfigFlags, flags args, printFlags *k8sget.PrintFlags) (int, error) {
+func loadConfig(configPath string) (conf Config, err error) {
+	viper.SetConfigFile(configPath)
+	viper.SetConfigType("yaml")
+
+	viper.AutomaticEnv()
+
+	err = viper.ReadInConfig()
+	if err != nil {
+		return
+	}
+
+	err = viper.Unmarshal(&conf)
+	return
+}
+
+func validateConfig(conf Config) error {
+	for i, exclusion := range conf.Exclusions {
+		if exclusion.Description == nil {
+			return fmt.Errorf("description must not be empty for exlusion rule %d", i)
+		}
+		if exclusion.Name == nil {
+			return fmt.Errorf("name must not be empty for exlusion rule %d", i)
+		}
+		if exclusion.Kind == nil {
+			return fmt.Errorf("kind must not be empty for exlusion rule %d", i)
+		}
+		if exclusion.Kind.Version == "" {
+			return fmt.Errorf("kind.version must not be empty for exlusion rule %d", i)
+		}
+		if exclusion.Kind.Kind == "" {
+			return fmt.Errorf("kind.kind must not be empty for exlusion rule %d", i)
+		}
+	}
+	return nil
+}
```

Comment:
main.go starts to get a bit out of hands here. It might  make sense to introduce a new pkg pkg/config.

What would even be nicer is that the config is versioned using kube style versioning, although this could also be introduced later.
```
apiVersion: gitops-zombies/v1
kind: Config
exclusions:
- .....
```

Reply:
definitely :+1: good catch !